tensorflow/core/common_runtime/dml/dml_execution_context.h

```diff
@@ -84,16 +85,45 @@ class DmlExecutionContextImpl {
  private:
   Microsoft::WRL::ComPtr<ID3D12Device> d3d_device_;
 
-  void SetCommandRecorder(DmlCommandRecorder* new_recorder);
-
   std::shared_ptr<DmlCommandQueue> queue_;
 
-  DmlCommandRecorder* current_recorder_ = nullptr;
+  bool closed_ = false;
 
-  // Up to one of these is active at a time
-  DmlCommandRecorder dml_recorder_;  // TODO: merge into this class
+  Microsoft::WRL::ComPtr<IDMLDevice> dml_device_;
+  Microsoft::WRL::ComPtr<IDMLCommandRecorder> recorder_;
 
-  bool closed_ = false;
+  // Descriptors are allocated from a pool. The current heap pointer is only
+  // used to avoid redundantly setting the same heap; it does not have ownership
+  // of the heap object.
+  DmlDescriptorPool descriptor_pool_;
+  ID3D12DescriptorHeap* current_descriptor_heap_ = nullptr;
+
+  DmlAllocator* allocator_;
```

Comment:
Nit: inconsistent initialization. It looks like this object initializes everything inline, so this should do so too.